src/inmanta/deploy/tasks.py

```diff
@@ -95,6 +95,18 @@ async def get_executor(
             agent_name=agent_name, agent_uri="NO_URI", code=code
         )
 
+        # Bail out if the current task's resource needs code
+        # that failed to load/install:
+        inmanta_module_name = self.resource.split("::")[0]
+        if inmanta_module_name in my_executor.failed_modules:
+            raise BaseExceptionGroup(
```

Comment:
This does not look right. `Base...` are not intended to be raised afaik.

I'm also not sure if it makes sense to report on all modules here, or only the one that failed, as we did before. @wouterdb wdyt?

Reply:
> Base... are not intended to be raised afaik

I couldn't find anything against it in the [docs](https://docs.python.org/3.13/library/exceptions.html#BaseExceptionGroup). I'm not very familiar with these, iirc it was suggested to me in a previous code review to use them [here](https://github.com/inmanta/inmanta-core/blob/9f4949d60a15306e6c7b849a1c77cf301876af5e/src/inmanta/server/protocol.py#L243)

Reply:
> or only the one that failed

Current behavior is that if the currently deploying resource's module's name is in the set of failed modules for the executor, we report all these failed modules. I think this is necessary if e.g. a loading failure occurs in a "dependent" file and not in the "main" module in which the resource lives.
The drawback is that in case of multiple, unrelated module loading failures, they would all be reported for each failing resource.

I don't have a good sense of which is best.

Reply:
> I think this is necessary if e.g. a loading failure occurs in a "dependent" file and not in the "main" module in which the resource lives.

I agree that this is an important consideration. Do you mean to say that if `x` fails because it tries to `import y`, you think that a failure that points to both is clearer than simply a failure that says that `x` failed to load?

Reply:
Yes

Reply:
I find it difficult to make a decision. I think there is something to say for either option. I'm fine with this.